src/main/diagnostics/diagnostics.ts

```diff
@@ -0,0 +1,48 @@
+import yaml from 'yaml';
+
+import DIAGNOSTICS_TABLE from '@/assets/diagnostics.yaml';
+
+type DiagnosticsFix = {
+  fixes: Record<'description', string>
+};
+
+type DiagnosticsCheck = {
+  id: string,
+  documentation: string,
+  description: string,
+  mute: boolean,
+  fixes: DiagnosticsFix,
+};
+
+type DiagnosticsCategories = {
```

Comment:
Does this need to be an independent type? It's only used once, and can easily be defined inline instead.

(I find inline to be slightly easier to read, given that the structures involved are so small.)

Reply:
I prefer naming things when defining a non-trivial taxonomy, even the small ones. And this thing is in flux anyway.